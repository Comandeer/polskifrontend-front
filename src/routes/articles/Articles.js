import React, { PropTypes } from 'react';
import withStyles from 'isomorphic-style-loader/lib/withStyles';
import styles from './Articles.styl';
import { connect } from 'react-redux';
import mapStateToProps from '../../core/redux/mapStateToProps';
import mapDispatchToProps from '../../core/redux/mapDispatchToProps';
import { decode } from 'he';
import HeaderSettings from '../../components/Layout/HeaderSettings';
import ArticleView from './parts/ArticleView';

class Articles extends React.Component {
  static propTypes = {
    context: PropTypes.object.isRequired
  };

  componentWillUnmount() {
    const { actions: { articlesClearDataLoaded } } = this.props;
    articlesClearDataLoaded();
  }

  render() {
    const { articlesState: { article, articleLoading }, context } = this.props;
    const title = `${article.title} | Polski Front-End`;
    const description = article.description ? decode(article.description.replace(/(<([^>]+)>)/ig, '')).substr(0, 140) + '...' : '';

    const blog = article._blog || {};

    return (
      <div className={styles.container}>
        <HeaderSettings description={description} title={title} currentPath={context.path} />
        <ArticleView blogName={blog.name || ''}
                     blogIcon={blog.favicon || ''}
                     blogHref={blog.href || ''}
                     date={article.date || ''}
                     description={article.description ? decode(article.description.replace(/(<([^>]+)>)/ig, '')).trim().substr(0, 600) + '...' : ''}
                     href={article.href || ''}
                     isLoading={articleLoading}
                     title={article.title || ''}
        />
      </div>
    );
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(withStyles(styles)(Articles));
