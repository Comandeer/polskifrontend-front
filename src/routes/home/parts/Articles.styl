@import "../../../styles/globals.styl"
@import "../../../styles/variables.styl"
@import "../../../styles/colors.styl"

.items
  background $color-secondary
  padding 5px $global-padding
  border-radius 2px
  min-height 167px

  .item
    border-bottom 1px dotted $color-tertiary
    border-left 6px solid $color-tertiary
    padding-left 10px
    margin 25px 5px

    &--today
      border-bottom 1px dotted $color-oposite
      border-left 6px solid $color-oposite

    &__new
      background $color-oposite
      color $color-secondary
      padding 5px $global-padding
      display none
      border-radius 5px
      font-weight normal
      font-size 12px
      vertical-align text-bottom
      margin-right 5px

      &--visible
        display inline-block

    &__link
      display block
      margin 7px 0
      text-decoration none
      font-size 20px
      font-weight bold
      font-family $font-header
      color $color-primary
      transition color $animation-time
      white-space: nowrap;
      overflow: hidden;
      text-overflow: ellipsis;
      &:hover
        color lighten($color-primary, 50%)

    .meta
      font-size 14px
      padding-top $global-margin
      color lighten($color-primary, 50%)

      &__description
        font-size 16px
        line-height 1.3em
        color $color-primary
        padding $global-padding 0
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;

        img
          width 25%
          height auto
          float left
