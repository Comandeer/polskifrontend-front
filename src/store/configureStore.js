/* eslint-disable no-undef */
import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';
import rootEpic from '../epics';
import createLogger from './logger';
import { createEpicMiddleware } from 'redux-observable';

export default function configureStore(initialState) {
  const middleware = [thunk, createEpicMiddleware(rootEpic)];

  let enhancer;

  if (__DEV__) {
    middleware.push(createLogger());

    // https://github.com/zalmoxisus/redux-devtools-extension#redux-devtools-extension
    let devToolsExtension = f => f;
    if (process.env.BROWSER && window.devToolsExtension) {
      devToolsExtension = window.devToolsExtension();
    }

    enhancer = compose(
      applyMiddleware(...middleware),
      devToolsExtension
    );
  } else {
    enhancer = applyMiddleware(...middleware);
  }

  // See https://github.com/rackt/redux/releases/tag/v3.1.0
  const store = createStore(rootReducer, initialState, enhancer);

  return store;
}
