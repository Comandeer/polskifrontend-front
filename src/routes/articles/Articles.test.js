/* eslint-disable */
import React from 'react';
import Articles from './Articles';
import renderer from 'react-test-renderer';
import StyleProvider from '../../../test/StyleProvider';
import configureStore from 'redux-mock-store';

describe('Articles component', () => {
  const mockBlog = {
    name: 'mock name',
    favicon: 'mock favicon',
    href: 'mock href',
    date: 'mock date'
  };
  const mockArticle = {
    title: 'mock title',
    description: 'mock description',
    blog: mockBlog,
    href: 'mock href',
    date: new Date('2018-01-01')
  };
  const mockState = {
    article: mockArticle,
    articleLoading: false
  };

  const mockStore = configureStore([]);

  it('renders correctly', () => {
    const tree = renderer
      .create(
        <StyleProvider insertCss={() => {}}>
          <Articles articlesState={mockState} context={{ path: 'mock path' }} store={mockStore(mockState)} />
        </StyleProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
