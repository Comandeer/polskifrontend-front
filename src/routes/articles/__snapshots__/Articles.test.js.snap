// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles component renders correctly 1`] = `
<div
  className="container"
>
  <div>
    <div
      className="container"
    >
      <div
        className="container container"
      >
        <div
          className="container"
        >
          <h2
            className="container__title"
          >
            <div>
              <span>
                
              </span>
            </div>
          </h2>
          <div
            className="wrapper"
          >
            <p
              className="wrapper__description"
            >
              
            </p>
            <div
              className="nav"
            >
              <a
                className="back"
                href="/"
                onClick={[Function]}
              >
                <i
                  className="icon-left-big"
                />
                Strona główna
              </a>
            </div>
            <div
              className="item"
            >
              <h2
                className="item__header"
              >
                <a
                  href="mock href"
                  rel="nofollow"
                  target="_blank"
                  title="mock title"
                >
                  mock title
                </a>
              </h2>
              <div
                className="meta"
              >
                <p
                  className="meta__date"
                >
                  01-01-2018
                </p>
                <pre
                  className="meta__message"
                >
                  mock description...
                </pre>
              </div>
              <div
                className="more"
              >
                <h3
                  className="more__header"
                >
                  Chcesz więcej? Przeczytaj w oryginale!
                </h3>
                <a
                  className="more__link"
                  href="mock href"
                  rel="nofollow"
                  target="_blank"
                  title="mock title"
                >
                  <i
                    className="icon-link-ext-alt"
                  />
                  Przejdź do artykułu
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
